ext/spl/spl_directory.c

```diff
@@ -183,14 +182,11 @@ static void spl_filesystem_object_free_storage(zend_object *object) /* {{{ */
    - clone
    - new
  */
-static zend_object *spl_filesystem_object_new_ex(zend_class_entry *class_type)
+static zend_object *spl_filesystem_object_new(zend_class_entry *class_type)
 {
 	spl_filesystem_object *intern;
 
-	intern = emalloc(sizeof(spl_filesystem_object) + zend_object_properties_size(class_type));
-	memset(intern, 0,
-		MAX(XtOffsetOf(spl_filesystem_object, u.dir.entry),
-			XtOffsetOf(spl_filesystem_object, u.file.escape) + sizeof(int)));
+	intern = ecalloc(1, sizeof(spl_filesystem_object) + zend_object_properties_size(class_type));
```

Comment:
I stumbled on this by accident. The `memset` was intentional to avoid writing the entirety of `spl_filesystem_object.u.dir.entry`. Did this cause any issues? If not, I think this should be reverted.

Reply:
Doesn't `ecalloc` take care of setting the whole memory to `0` tho?

Reply:
Yes, but we don't _need_ to do that. `spl_filesystem_object.u.dir.entry` will later be initialized anyway.

Reply:
Ah right, I personally find the memset confusing, which is why I replaced it with a calloc. I don't _really_ care but then SPL is convoluted enough that being clear seems like a positive to me.

Reply:
A comment should suffice :) I'll take care of it. Thanks!